hazelcast/src/main/java/com/hazelcast/internal/partition/ChunkSuppliers.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.partition;
+
+import com.hazelcast.internal.util.CollectionUtil;
+import com.hazelcast.spi.impl.operationservice.Operation;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+
+import static com.hazelcast.internal.util.Preconditions.checkNoNullInside;
+import static com.hazelcast.internal.util.Preconditions.checkTrue;
+import static com.hazelcast.internal.util.Preconditions.isNotNull;
+
+/**
+ * Helper methods to create various {@link ChunkSupplier} objects
+ */
+public final class ChunkSuppliers {
+
+    private ChunkSuppliers() {
+    }
+
+    /**
+     * Intended use cases:
+     * <ul>
+     * <li>When migration data fits in one single chunk</li>
+     * <li>When a service doesn't support data migration in chunks</li>
+     * </ul>
+     */
+    public static ChunkSupplier newSingleChunkSupplier(Supplier<Operation> operationSupplier) {
+        isNotNull(operationSupplier, "operationSupplier");
+
+        return new SingleChunkSupplier(operationSupplier);
+    }
+
+    private static final class SingleChunkSupplier implements ChunkSupplier {
+
+        private final Supplier<Operation> operationSupplier;
+
+        private boolean hasMoreChunks = true;
+
+        private SingleChunkSupplier(Supplier<Operation> operationSupplier) {
+            this.operationSupplier = operationSupplier;
+        }
+
+        @Nullable
+        @Override
+        public Operation next() {
```

Comment:
the provided supplier is invoked only once, and from usages it seems there is no null returns for now. I was wondering under what circumstances will the `next()` call return null

Reply:
I have added that possible case as a javaDoc. Please see it.

Reply:
thank you :)